src/services/mongoose/users.ts

```diff
@@ -1,60 +1,86 @@
-import {
-  CouldNotFindUserFromIdError,
-  CouldNotFindUserFromPhoneError,
-  RepositoryError,
-} from "@domain/errors"
-import { User } from "@services/mongoose/schema"
+import { getLocale } from "@config"
+import { CouldNotFindError, RepositoryError } from "@domain/errors"
 
-import { fromObjectId, toObjectId, parseRepositoryError } from "./utils"
+import { User } from "./schema"
 
-export const UsersRepository = (): IUsersRepository => {
-  const findById = async (userId: UserId): Promise<User | RepositoryError> => {
-    try {
-      const result = await User.findOne({ _id: toObjectId<UserId>(userId) }, projection)
+import { parseRepositoryError } from "./utils"
 
-      if (!result) {
-        return new CouldNotFindUserFromIdError(userId)
-      }
+export const translateToUser = (user: UserRecord): User => {
+  const language = (user?.language ?? "") as UserLanguageOrEmpty
+  const languageOrDefault = language === "" ? getLocale() : language
+  const deviceTokens = user.deviceTokens ?? []
+  const phoneMetadata = user.phoneMetadata
+  const phone = user.phone
+
+  return {
+    id: user.userId as KratosUserId,
+    language: language,
+    languageOrDefault: languageOrDefault as UserLanguage,
+    deviceTokens: deviceTokens as DeviceToken[],
+    phoneMetadata,
+    phone,
+  }
+}
+
+export const UsersRepository = (): IUserRepository => {
+  const findById = async (id: KratosUserId): Promise<User | RepositoryError> => {
+    try {
+      const result = await User.findOne({ userId: id })
+      if (!result) return translateToUser({ userId: id, deviceTokens: [] })
 
-      return userFromRaw(result)
+      return translateToUser(result)
     } catch (err) {
+      //  else
       return parseRepositoryError(err)
     }
   }
 
   const findByPhone = async (phone: PhoneNumber): Promise<User | RepositoryError> => {
     try {
-      const result = await User.findOne({ phone }, projection)
-      if (!result) {
-        return new CouldNotFindUserFromPhoneError(phone)
-      }
+      const result = await User.findOne({ phone })
+      if (!result) return new CouldNotFindError()
 
-      return userFromRaw(result)
+      return translateToUser(result)
+    } catch (err) {
+      //  else
+      return parseRepositoryError(err)
+    }
+  }
+
+  const list = async (): Promise<User[] | RepositoryError> => {
```

Comment:
Think this is probably not a good idea to expose a way to get all users. At least we should return an iterater so as to not blow up memory.